lib/wasi/src/state/env.rs

```diff
@@ -694,6 +718,60 @@ impl WasiEnv {
         }
     }
 
+    pub(crate) fn process_wakes_internal(
+        ctx: &mut FunctionEnvMut<'_, Self>,
+    ) -> Result<(), WasiError> {
+        let env = ctx.data();
+        let state = env.state();
+
+        // Invoke the callback for each occurance of a wake
+        for (id, woken) in state.wakers.pop_wakes() {
+            Self::process_one_wake_internal(ctx, id, woken)?;
+        }
+        Ok(())
+    }
+
+    pub(crate) fn process_one_wake_internal(
+        ctx: &mut FunctionEnvMut<'_, Self>,
+        id: WakerId,
+        woken: bool,
+    ) -> Result<(), WasiError> {
+        let env = ctx.data();
+        let inner = env
+            .try_inner()
+            .ok_or_else(|| WasiError::Exit(Errno::Fault.into()))?;
+
+        let handler = match woken {
+            true => inner.waker_wake.clone(),
+            false => inner.waker_drop.clone(),
+        };
+        if let Some(handler) = handler {
+            let high = id / u32::MAX as WakerId;
```

Comment:
This doesn't look like a proper 64bits -> high/low 32bits extraction (because 0x1_0000
_0000 will give 1 / 1 instead of 1 / 0). Is that on purpose?

Reply:
Dropped the support for 64bits